singlelayerqg.jl

```diff
@@ -245,7 +252,10 @@ Update the variables in `vars` with the solution in `sol`.
 """
 function updatevars!(sol, vars, params, grid)
   @. vars.zetah = sol
-  @. vars.psih  = -grid.invKrsq * vars.zetah
+  @. vars.psih  = - vars.zetah / (grid.Krsq + params.kdef^2)
+  if params.kdef == 0.0
```

Comment:
adding an `if` statement here might have ramifications in performance...

the more I think about it I lean towards defining a `streamfunctionfrompv!()` function and overload it, e.g.:

```julia
function streamfunctionfrompv!(vars, params::BarotropicParams, grid)
  @. vars.psih =  - grid.invKrsq * vars.sol
end

function streamfunctionfrompv!(vars, params::EquivalentBarotropicParams, grid)
  @. vars.psih  = - vars.sol / (grid.Krsq + params.kdef^2)
end
```

@glwagner, what do you reckon?

Reply:
4650164 and 049bad0 resolve this.